tests/integration/test_store.py

```diff
@@ -42,6 +39,140 @@ def test_legacy_store(mini_sentry, relay_chain):
     assert mini_sentry.get_hits("/api/0/relays/projectids/") == 1
 
 
+@pytest.mark.parametrize(
+    "filter_config, should_filter",
+    [
+        ({"releases": {"releases": ["1.2.2", "1.2.3"]}}, True),
+        ({"releases": {"releases": ["1.2.2", "1.2.4"]}}, False),
+        ({"legacyBrowsers": {"options": ["ie9", "ie10"], "isEnabled": True}}, True),
+        ({"legacyBrowsers": {"options": ["ie9"], "isEnabled": True}}, False),
+        ({"errorMessages": {"patterns": ["Panic: originalCreateNotification"]}}, True),
+        ({"errorMessages": {"patterns": ["Warning"]}}, False),
+        ({"clientIps": {"blacklistedIps": ["127.0.0.1"]}}, True),
+        ({"clientIps": {"blacklistedIps": ["122.33.230.14"]}}, False),
+        ({"localhost": {"isEnabled": True}}, True),
+        ({"localhost": {"isEnabled": False}}, False),
+        ({"browserExtensions": {"isEnabled": True}}, True),
+        ({"browserExtensions": {"isEnabled": False}}, False),
```

Comment:
Is it really worth having that many tests here with processing mode? This basically just tests that the config is parsed, the rest should already be unit tested. Also, it's quite slow. 

I'm wondering if this would make more sense in Sentry's Relay tests to also assert serialization and deserialization are compatible.

Reply:
I agree with this. Having this many tests is way too slow to test at this level. I would pick out 2-3 parameter configurations and throw the test away (ensuring it's unittested in Rust though)

Reply:
I absolutely agree that it is painfully slow (especially the tests that check that there is no message coming by waiting to time-out). This is the reason I wanted to port the approach we use in Sentry (which still dose not fix the tests that do not produce events). 

I think this is the minimum amount of tests one per filter. I have a negative test to prove that without the testing condition we have behaviour X (i.e. the events pass) and then exactly with the same setup except the testing condition we have behaviour Y (i.e. the event is filtered).  What I'm basically testing is that the filter is used in the event processing pipeline (I'm not unit testing the details of the filter functionality). A unit test would not really test this.

Reply:
I think even invoking a single filter (enabled) would be enough of a test, since this ensures that the config is passed on: 
- All other permutations are tested in unit tests, given that the config is passed in a single struct
- There need to be E2E tests in Sentry anyway to ensure that the filter names actually match what Sentry outputs. We had tests like these in the past already where we checked in with a typo and just didn't recognize.
- The filter _disabled_ tests are implicitly run by other tests that just ingest a regular event, since those also don't run any filters.